src/brsTypes/components/Regex.ts

```diff
@@ -0,0 +1,153 @@
+        super("roRegex");
+        this.jsRegex = new RegExp(expression.value, flags.value);
+
+        this.registerMethods([
+            this.isMatch,
+            this.match,
+            this.replace,
+            this.replaceAll,
+            this.split,
+            this.matchAll
+        ]);
+    }
+
+    toString(parent?: BrsType): string {
+        return "<Component: roRegex>";
+    }
+
+    private parseReplacementPattern(pattern: string): string {
+        return pattern.replace(/\\/g, "\$");
+    }
+
+    private isMatch = new Callable(
+        "ismatch",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.Boolean
+            },
+            impl: (interpreter: Interpreter, str: BrsString) => {
+                return this.jsRegex.test(str.value) ? BrsBoolean.True : BrsBoolean.False;
+            }
+        }
+    );
+
+    private match = new Callable(
+        "match",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.Object
+            },
+            impl: (interpreter: Interpreter, str: BrsString) => {
+                const result = this.jsRegex.exec(str.value);
+                let arr = [];
+                if (result !== null) {
+                    for (let i = 0; i < result.length; i++) {
+                        let item = result[i] ? new BrsString(result[i]) : BrsInvalid.Instance;
+                        arr.push(item);
+                    }
+                }
+
+                return new BrsArray(arr);
+            }
+        }
+    );
+
+    private replace = new Callable(
+        "replace",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String),
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.String
+            },
+            impl: (interpreter: Interpreter, str: BrsString, replacement: BrsString) => {
+                let replacementPattern = this.parseReplacementPattern(replacement.value);
+                const newStr = this.jsRegex[Symbol.replace](str.value, replacementPattern);
+                return new BrsString(newStr);
+            }
+        }
+    );
+
+    private replaceAll = new Callable(
+        "replaceall",
+        {
+            signature: {
+                args: [
+                    new StdlibArgument("str", ValueKind.String),
+                    new StdlibArgument("str", ValueKind.String)
+                ],
+                returns: ValueKind.String
+            },
+            impl: (interpreter: Interpreter, str: BrsString, replacement: BrsString) => {
+                const source = this.jsRegex.source;
+                const flags = this.jsRegex.flags + "g";
```

Comment:
If someone passes "g" into roRegex I think a second one will be applied here, which will raise a SyntaxError.  

    new RegExp("some thing", "gg"); // throws error

I'm not sure how Roku's implementation, but in both the constructor above and here we might want to check for duplicates.

Reply:
Great catch, @strattonbrazil!  I agree.  Maybe something like `this.jsRegex.flags.indexOf("g") === -1 ? this.jsRegex.flags + "g" : this.jsRegex.flags`?  Or something less verbose 🤷‍♂️

Reply:
I'm solving this with a different approach: avoiding the constructor to take unwanted flags. Since `g` is not a valid brighstcript flag it will be never duplicated.